datafusion/core/src/execution/context.rs

```diff
@@ -2175,13 +2175,13 @@ mod tests {
     async fn create_variable_err() -> Result<()> {
         let ctx = SessionContext::new();
 
-        let err = plan_and_collect(&ctx, "SElECT @=   X#=?!~ 5")
```

Comment:
without chis change, sqlparser complains (reasonably) about `X#=?!~ 5` not being a valid identifier. 

https://github.com/apache/arrow-datafusion/actions/runs/5621460600/job/15232281958?pr=7051

```
---- execution::context::tests::create_variable_err stdout ----
thread 'execution::context::tests::create_variable_err' panicked at 'assertion failed: `(left == right)`
  left: `"SQL error: ParserError(\"Expected end of statement, found: =\")"`,
 right: `"Error during planning: variable [\"@\"] has no type information"`', datafusion/core/src/execution/context.rs:2182:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```

So I changed the test to preserver the error (rather than updating the error to be a parse error)